src/DependencyInjection/SonataClassificationExtension.php

```diff
@@ -133,11 +134,15 @@ private function registerSonataDoctrineMapping(array $config): void
                 ->addOrder('position', 'ASC')
         );
 
+        $categoryCascade = ['persist', 'refresh', 'detach'];
+        if (class_exists(EntityManager::class) && method_exists(EntityManager::class, 'merge')) { // @phpstan-ignore-line
```

Comment:
You will lose the merge for ODM user.
https://github.com/doctrine/mongodb-odm/blob/2.10.x/lib/Doctrine/ODM/MongoDB/DocumentManager.php

it should be
```
if (
    class_exists(EntityManager::class) && method_exists(EntityManager::class, 'merge')
    || class_exists(DocumentManager::class) && method_exists(DocumentManager::class, 'merge')
)
```
but I assume it will create issue when both lib are installed.

So we have to know which entity we extends.

Maybe 
```
$classParents = class_parents($config['class']['category']);

if (\in_array(Entity/BaseCategory::class, $classParents, true) && class_exists(EntityManager::class) && !method_exists(EntityManager::class, 'merge')) {
     unset($categoryCascade['merge']);
}
```
?

Reply:
Sorry for late reply. I've checked odm src and it seems like `merge` isn't deprecated (even in 3.0 branch), so 2nd option should be just fine.